pkg/controller/secret/secret_controller.go

```diff
@@ -89,6 +73,184 @@ type ReconcileSecret struct {
+	receiverMakeItWarning = "make-it-warning"
+
+	// anything routed to "pagerduty" will alert/notify SREP
+	receiverPagerduty = "pagerduty"
+
+	// anything going to Dead Man's Snitch (watchdog)
+	receiverWatchdog = "watchdog"
+
+	// the default receiver used by the route used for pagerduty
+	defaultReceiver = receiverNull
+
+	// global config for PagerdutyURL
+	pagerdutyURL = "https://events.pagerduty.com/v2/enqueue"
+)
+
+func getPagerdutyRoute() *alertmanager.Route {
+	// order matters.
+	// these are sub-routes.  if any matches it will not continue processing.
+	// 1. route anything we want to silence to "null"
+	// 2. route anything that should be a warning to "make-it-warning"
+	// 3. route anything we want to go to PD
+	pagerdutySubroutes := []*alertmanager.Route{
+		// https://issues.redhat.com/browse/OSD-1966
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "KubeQuotaExceeded"}},
+		// https://issues.redhat.com/browse/OSD-2382
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "UsingDeprecatedAPIAppsV1Beta1"}},
+		// https://issues.redhat.com/browse/OSD-2382
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "UsingDeprecatedAPIAppsV1Beta2"}},
+		// https://issues.redhat.com/browse/OSD-2382
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "UsingDeprecatedAPIExtensionsV1Beta1"}},
+		// https://issues.redhat.com/browse/OSD-2980
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "CPUThrottlingHigh", "container": "registry-server"}},
+		// https://issues.redhat.com/browse/OSD-3008
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "CPUThrottlingHigh", "container": "configmap-registry-server"}},
+		// https://issues.redhat.com/browse/OSD-3010
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "NodeFilesystemSpaceFillingUp", "severity": "warning"}},
+		// https://issues.redhat.com/browse/OSD-2611
+		{Receiver: receiverNull, Match: map[string]string{"namespace": "openshift-customer-monitoring"}},
+		// https://issues.redhat.com/browse/OSD-3220
+		{Receiver: receiverNull, Match: map[string]string{"alertname": "SLAUptimeSRE"}},
+
+		// https://issues.redhat.com/browse/OSD-1922
+		{Receiver: receiverMakeItWarning, Match: map[string]string{"alertname": "KubeAPILatencyHigh", "severity": "critical"}},
+
+		// general: route anything in core namespaces to PD
+		{Receiver: receiverPagerduty, MatchRE: map[string]string{"namespace": alertmanager.PDRegex}},
+		// fluentd: route any fluentd alert to PD
+		// https://issues.redhat.com/browse/OSD-3326
+		{Receiver: receiverPagerduty, Match: map[string]string{"job": "fluentd"}},
+		{Receiver: receiverPagerduty, Match: map[string]string{"alertname": "FluentdNodeDown"}},
+		// elasticsearch: route any ES alert to PD
+		// https://issues.redhat.com/browse/OSD-3326
+		{Receiver: receiverPagerduty, Match: map[string]string{"cluster": "elasticsearch"}},
```

Comment:
I'm not familiar with this `cluster` parameter from the alert being used like that. Are we sure that's how we filter for elasticsearch related alerts?

Reply:
Need something to tag on.  All the alerts coming from elasticsearch include the label `cluster` with a value of `elasticsearch` so I picked that.  I think other common labels.  Ideally we'd get a `namespace` label and be done but we don't get it.  If this is a problem I can look again at the alerts.

Reply:
If this label is assigned to all `elasticsearch` alerts +1